### PR TITLE
SIMSBIOHUB-614: Fix file upload error handling.

### PR DESCRIPTION
**Pull request by NickPhura**
## Links to Jira Tickets

https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-614?

## Description of Changes

Split the project/survey report/non-report attachment dialogs into 2 components.
Made 2 new attachment dialogs - one for single items, one for many items.
Added error handling to catch file upload errors.
Moved a few files around to group like-components.

## Testing Notes

Changes to Project/Survey Report and Non-Report attachment upload
- Should work as before, and should handle any errors thrown during upload.

Changes to observation/telemetry/sample-site/animal csv import
- Should work as before, and should handle any errors thrown during upload.

### CODE REVIEW

File: app/src/components/attachments/FileUploadWithMeta.tsx [hunk 1]
```diff
@@ -1,84 +1,52 @@
...
-  fileHandler?: IFileHandler;
-  onSuccess?: IOnUploadSuccess;
-}
+export const FileUploadWithMeta = () => {
+  const { handleSubmit, setFieldValue, setFieldError, values, errors } = useFormikContext<IReportMetaForm>();
 
-export const FileUploadWithMeta: React.FC<IFileUploadWithMetaProps> = (props) => {
-  const { handleSubmit, setFieldValue, errors } = useFormikContext<IReportMetaForm>();
+  const onStatus = (status: UploadFileStatus) => {
+    console.log(status);
```
Comment (mauberti-bc) [inline app/src/components/attachments/FileUploadWithMeta.tsx]: Deliberate console.log?

**Review by mauberti-bc — DISMISSED** [review]
Tested locally and works well 👍 

Minor future UI improvement: Closing the error dialog should also close the upload attachment dialog to avoid displaying "Complete" or "Success" on the upload dialog despite an error.

image

File: app/src/components/attachments/FileUploadWithMeta.tsx (hunk 1 above)
Comment (NickPhura) [reply app/src/components/attachments/FileUploadWithMeta.tsx]: Nope! will remove

**Comment by NickPhura** [comment]
@mauberti-bc 
> Minor future UI improvement: Closing the error dialog should also close the upload attachment dialog to avoid displaying "Complete" or "Success" on the upload dialog despite an error.

The error popup was actually a copy/paste mistake. The file upload component handles individual file errors already, so I just had to remove the try/catch and let it bubble up:

image

Tweaked some other trivial stuff as well, and added some JSDoc. So should all be happy now.

**Review by mauberti-bc — APPROVED** [review]
👍

**Pull request merged**